### PR TITLE
Task 119339: Change log level [TAGS NOT PUSHED] AB#119339

### PR DESCRIPTION
**Pull request by esepato**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "express-nemo-bootstrap",
-  "version": "1.5.0",
+  "version": "1.6.0",
```
Comment (ohritz) [inline package.json:3]: Maybe not "waste" a minor version, and 1.5.1 is sufficient?

File: logger.js [hunk 2]
```diff
@@ -62,10 +62,8 @@ const log = (data, level, context) => {
     message = data ?? ''
   } else {
     message = data.message ?? ''
-    logContext = extend({}, logContext, data)
-    delete logContext.message
-    delete logContext.level
-    delete logContext.timestamp
+    const { message, level, timestamp, ...dataProps } = data
+    logContext = extend(dataProps, logContext)
```
Comment (ohritz) [inline logger.js:66]: Nice!

File: logger.js [hunk 3]
```diff
@@ -50,7 +50,7 @@ const pinoLogger = pino({
 
 const log = (data, level, context) => {
   let logContext = {
-    context: extend({}, defaultStructure.context, context)
+    context: extend(defaultStructure.context, context)
```
Comment (ohritz) [inline logger.js:53]: when not giving an empty object as the target, it seems the `defaultStructure.context` will be a reference and any unfortunate mutations to the `logContext.context` will mutate the `defaultStructure.context`.
I had to verify in a jsfiddle: https://jsfiddle.net/ohreo/ydfo21pa/3/#&togetherjs=14k2wvqnot

**Review by ohritz — CHANGES_REQUESTED** [review]

File: logger.js (hunk 3 above)
Comment (esepato) [reply logger.js:53]: Oh, wow. Good catch, putting the empty object back.

File: package.json (hunk 1 above)
Comment (esepato) [reply package.json:3]: No big deal for me. I thought that as a consumer of the package, you will know that something is breaking with the minor, and in this case it is the logging behaviour.

**Comment by esepato** [comment]
> In `middlewares.js` `requestResponseLogger` is created from another npm package (expressHttpContextRequestResponseLogger) which is "ours" the default options object there looks like this:
> 
> ```js
> const defaults = {
>   loggerFunctionName: 'debug',
>   logEventFactory: (req, res) =>
>     `${req.method} ${req.url} - HTTP ${res.statusCode}`
> }
> ```
> 
> With the change from default loglevel being set to 'info' all request loging will be ignored/filtered out. Is that what we want?

I assume not! I saw that some request logs actually were removed. Nice catch.

So ideally we would like to use the same as default? The request logging doesn't seem to be optional, it will always be used and currently always use the default level since this solution doesn't provide any way to override it.

**Comment by ohritz** [comment]
> > In `middlewares.js` `requestResponseLogger` is created from another npm package (expressHttpContextRequestResponseLogger) which is "ours" the default options object there looks like this:
> > ```js
> > const defaults = {
> >   loggerFunctionName: 'debug',
> >   logEventFactory: (req, res) =>
> >     `${req.method} ${req.url} - HTTP ${res.statusCode}`
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > With the change from default loglevel being set to 'info' all request loging will be ignored/filtered out. Is that what we want?
> 
> I assume not! I saw that some request logs actually were removed. Nice catch.
> 
> So ideally we would like to use the same as default? The request logging doesn't seem to be optional, it will always be used and currently always use the default level since this solution doesn't provide any way to override it.

I think you can just set in express-nemo-bootstrap:middleware.js 
```js
const requestResponseLogger = expressHttpContextRequestResponseLogger({
    logEventFactory: logEventFactory.createRequestResponseLogEvent,
    loggerFunctionName: "info"
  })
```

we are already overiding the default logEventFactory.

File: logger.js [hunk 4]
```diff
@@ -62,10 +62,8 @@ const log = (data, level, context) => {
     message = data ?? ''
   } else {
     message = data.message ?? ''
-    logContext = extend({}, logContext, data)
-    delete logContext.message
-    delete logContext.level
-    delete logContext.timestamp
+    const { message, level, timestamp, ...dataProps } = data
```
Comment (ohritz) [inline logger.js:65]:
```suggestion
    // eslint-disable-next-line no-unused-vars
    const { message, level, timestamp, ...dataProps } = data
```

Just add this, I get an error in vs-code

**Pull request merged**